src/EditorFeatures/CSharpTest/UseCompoundAssignment/UseCompoundCoalesceAssignmentTests.cs

```diff
@@ -110,6 +110,29 @@ public async Task TestNoSideEffects2()
 {
     private string goo;
     private string Goo => this.goo ??= new string('c', 42);
+}");
+        }
+
+        [WorkItem(38059, "https://github.com/dotnet/roslyn/issues/38059")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestNullableValueType()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void Goo()
+    {
+        int? a = null;
+        var x = a [||]?? (a = 1);
+    }
+}",
+@"class Program
+{
+    void Goo()
+    {
+        int? a = null;
+        var x = a ??= 1;
```

Comment:
To be clear, this is changing the type of `var`. Is this ok? Should the IDE be making the `var` `int?` here, as it's changing the semantic meaning of the code? What about a scenario like this:

```cs
using System;
class C
{
    static void M(int a) => Console.WriteLine("Called with int");
	static void M(int? a) => Console.WriteLine("Called with int?");
	static void Main()
	{
		int? a = null;
		var b = (a ?? (a = 1));
		M(b);
	}
}
```

I would expect that the second overload should still be called after the refactoring is applied.

Reply:
Handled this case now. :)